tests/FileHelperTest.php

```diff
@@ -220,16 +272,501 @@ public function testNormalizePath(): void
         // should not be touched.
         // https://msdn.microsoft.com/en-us/library/windows/desktop/aa365247%28v=vs.85%29.aspx
         // https://github.com/yiisoft/yii2/issues/13034
-        $this->assertEquals('\\\\server/share/path/file', FileHelper::normalizePath('\\\\server\share\path\file', '\\'));
+        $this->assertEquals('\\\\server/share/path/file', FileHelper::normalizePath('\\\\server\share\path\file'));
+    }
+
+    /**
+     * Copy directory.
+     *
+     * @depends testCreateDirectory
+     *
+     * @return void
+     */
+    public function testCopyDirectory(): void
+    {
+        $source = 'test_src_dir';
+        $files = [
+            'file1.txt' => 'file 1 content',
+            'file2.txt' => 'file 2 content',
+        ];
+
+        $this->createFileStructure([
+            $source => $files,
+        ]);
+
+        $basePath = $this->testFilePath;
+        $source = $basePath . '/' . $source;
+        $destination = $basePath . '/test_dst_dir';
+
+        FileHelper::copyDirectory($source, $destination);
+
+        $this->assertFileExists($destination, 'Destination directory does not exist!');
+
+        foreach ($files as $name => $content) {
+            $fileName = $destination . '/' . $name;
+            $this->assertFileExists($fileName);
+            $this->assertStringEqualsFile($fileName, $content, 'Incorrect file content!');
+        }
+    }
+
+    /**
+     * Copy directory recursive.
+     *
+     * @return void
+     */
+    public function testCopyDirectoryRecursive(): void
+    {
+        $source = 'test_src_dir_rec';
+        $structure = [
+            'directory1' => [
+                'file1.txt' => 'file 1 content',
+                'file2.txt' => 'file 2 content',
+            ],
+            'directory2' => [
+                'file3.txt' => 'file 3 content',
+                'file4.txt' => 'file 4 content',
+            ],
+            'file5.txt' => 'file 5 content',
+        ];
+
+        $this->createFileStructure([
+            $source => $structure,
+        ]);
+
+        $basePath = $this->testFilePath;
+        $source = $basePath . '/' . $source;
+        $destination = $basePath . '/test_dst_dir';
+
+        FileHelper::copyDirectory($source, $destination);
+
+        $this->assertFileExists($destination, 'Destination directory does not exist!');
+
+        $checker = function ($structure, $dstDirName) use (&$checker) {
+            foreach ($structure as $name => $content) {
+                if (is_array($content)) {
+                    $checker($content, $dstDirName . '/' . $name);
+                } else {
+                    $fileName = $dstDirName . '/' . $name;
+                    $this->assertFileExists($fileName);
+                    $this->assertStringEqualsFile($fileName, $content, 'Incorrect file content!');
+                }
+            }
+        };
+        $checker($structure, $destination);
+    }
+
+    /**
+     * Copy directory not recursive.
```

Comment:
Such phpdoc is not useful so could be removed.

Reply:
As for the functions if I would like to leave the header, because that is how we know what each function does, it is informative that you think or I delete it.

Reply:
In this particular case and many others the phpdoc is exactly the same as method name so it doesn't add much.